pkg/etag/etag.go

```diff
@@ -176,6 +178,50 @@ var _ Tagger = ETag{} // compiler check
 // the Tagger interface.
 func (e ETag) ETag() ETag { return e }
 
+// FromContentMD5 decodes and returns the Content-MD5
+// as ETag, if set. If no Content-MD5 header is set
+// it returns an empty ETag and no error.
+func FromContentMD5(h http.Header) (ETag, error) {
+	v, ok := h["Content-Md5"]
+	if !ok {
+		return nil, nil
+	}
+	if v[0] == "" {
+		return nil, errors.New("etag: content-md5 is set but contains no value")
+	}
+	b, err := base64.StdEncoding.DecodeString(v[0])
```

Comment:
You should be using Strict() here

Reply:
Wouldn't that may break clients that send content-md5 values with non-zero padding characters?
Also, it does not prevent e.g. sending a hex value as content-md5: `fmt.Println(base64.StdEncoding.Strict().DecodeString("d41d8cd98f00b204e9800998ecf8427e"))`

In any case, `len(b) != md5.Size` catches all of that.

Reply:
Correct they can never send non-base64 @aead without padding - all AWS SDKs send correct base64 values.

Reply:
Ok 👍